### DISCUSSION TITLE
Could not load file or assembly System.Core

### DISCUSSION THREAD

**Discussion by ghost** [jbevain/cecil#796] [Q&A] [answered]
When running `mono.cecil` and `mono.cecil.cil` on .Net framework projects built on .Net Framework 3.5-4.7 I continue to receive the following errors after I `.Write()` the modified file out to disk (see below).
Some of the modifications include base64 encoding strings in EXE, add 1 new TypeDefinition, and 3 new MethodDefinition. None of the projects use System.Core before touched by cecil. The issue was first observed to have started about a month ago. Role back of cecil from nuget to previous version appears to not resolve issue either. Issue did not exist for certain 6 months ago. Once I review the code in dnspy i can see the reference added. Is there a way to stop the addition of System.Core to the projects?

Error output after crash:
`Could not load file or assembly 'System.Core, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' or one of its dependencies. The system cannot find the file specified.`

**Comment by jbevain** [comment, marked as answer]
Hi!

Thanks for reaching out. Unfortunately it's hard for me to say what's going with little information about your project or assemblies. Your best bet at this point is to build your own Cecil and debug what's going on.

Thanks!

**Comment by Zastai** [comment]
Chances are, you are adding type references into the assembly created from `System.Type` instances. That will cause references to be added to the assembly that defined those types _in your running code_.

So if you use `Guid.GetType()` to add a `TypeReference` from code that runs under .NET Framework 4.8 (or even .NET Core 3.1), it will cause a reference to the corresponding defining assembly, even if that is not the appropriate one to use for the target framework of the assembly you are editing.q
